src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
Ah I see what you want to achieve here. You want to inject your extended version of EventHubQueueCache, but still reuse the set up logic on the cache, such as time predicate etc. 

And also it can be used for use case when application wants to use default set up on the EventHubQueueCache, but just add some adjustment on top of that, such as add more pressure monitor,etc. 

I can see another alternative which might be less confusing for users:  instead of making this `CreateCache` a method, we can  make it a Func, just like CreateCacheFactory func. 

Another alternative which I like better. Since we are intending to add more extension point to `CreateCacheFactory` . Maybe it's a good idea to encapsulate all those functionality into a CreateCacheFactory class, and user can extend that CreateCacheFactory class by inheritance or decorating. This class will be responsible for create EventHubQueueCache and configure it. 


``